common/core/src/main/java/org/kaaproject/kaa/server/common/core/algorithms/AvroUtils.java

```diff
@@ -92,4 +92,38 @@ public static void copyJsonProperties(JsonProperties src, JsonProperties dst) {
         }
     }
 
+
+    public static JsonNode injectUuids(JsonNode json) {
+        boolean containerWithoutId = json.isContainerNode() && !json.has(UUID_FIELD);
+        boolean notArray = !json.isArray();
+        boolean childIsNotArray = !(json.size() == 1 && json.getElements().next() instanceof ArrayNode);
+
+        if (containerWithoutId && notArray && childIsNotArray) {
+            ((ObjectNode)json).put(UUID_FIELD, (Integer)null);
+        }
+
+        for (JsonNode node : json) {
+            if (node.isContainerNode())
+                injectUuids(node);
```

Comment:
I am not sure that this is method has correct API. Since you ignore the result of the method here, I believe it is better to return void. This will clarify behaviour of the method to the users.